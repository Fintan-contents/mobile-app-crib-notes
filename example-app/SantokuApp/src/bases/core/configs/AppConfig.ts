import Constants from 'expo-constants';
import {Platform} from 'react-native';

/**
 * 環境設定値を取得できます。
 *
 * ```ts
 * import {AppConfig} from 'bases';
 *
 * console.log(AppConfig.appNameHome);
 * ```
 *
 * {@link Constants.expoConfig.extra}.XXXで取得できる値が、型定義ではanyとなっています。
 * しかし、環境設定値はapp.config.xxx.jsに必ず適切な型で設定されている想定のため、type assertionしています。
 *
 * app.config.xxx.jsに環境設定値が設定されていない場合や、適切な型で設定されていない場合は、実行時にエラーとなります。
 */
export abstract class AppConfig {
  static get termsUrl(): string {
    return Constants.expoConfig?.extra?.termsUrl as string;
  }

  static get santokuAppBackendUrl(): string {
    const url = Constants.expoConfig?.extra?.santokuAppBackendUrl as string;
    return Platform.OS === 'android' ? url.replace('localhost', '10.0.2.2') : url;
  }

  static get requestTimeout(): number | undefined {
    const timeout = Constants.expoConfig?.extra?.requestTimeout as number;
    return isNaN(timeout) ? undefined : timeout;
  }

  static get storeUrl(): string | undefined {
    return Platform.select({
      ios: Constants.expoConfig?.extra?.appStoreAppUrl as string,
      android: Constants.expoConfig?.extra?.googlePlayAppUrl as string,
    });
  }

  static get mobileAppCribNotesWebsiteUrl(): string {
    return Constants.expoConfig?.extra?.mobileAppCribNotesWebsiteUrl as string;
  }

  static get mobileAppCribNotesRepositoryUrl(): string {
    return Constants.expoConfig?.extra?.mobileAppCribNotesRepositoryUrl as string;
  }

  static get mswEnabled(): boolean {
    return Constants.expoConfig?.extra?.mswEnabled as boolean;
  }
}
