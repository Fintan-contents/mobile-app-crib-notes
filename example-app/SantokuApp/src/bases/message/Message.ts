/**
 * Copyright 2023 TIS Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import {handleError} from 'bases/core/errors/handleError';

import {MessageKey} from './BundledMessages';
import {RuntimeError} from '../core/errors/RuntimeError';

/**
 * メッセージをロードします。
 */
interface MessagesLoader {
  load(): Promise<Record<MessageKey, string>>;
}

let cache: Record<MessageKey, string> | undefined;

/**
 * メッセージをロードします。
 * @param loader メッセージをロードするクラス
 */
async function loadMessages(loader: MessagesLoader) {
  cache = await loader.load();
}

/**
 * メッセージを取得します。
 * {@link loadMessages}を呼び出してして、メッセージをロード後に使用してください。
 * @param key メッセージのキー
 * @param options メッセージのオプション
 * @returns 指定されたメッセージキーに該当するメッセージ。指定されたメッセージキーに該当するメッセージが存在しない場合はメッセージキーを返却
 *
 * @example
 * 定義されているメッセージの例
 * ```
 * export const BundledMessages = {
 *  'validation.email': 'メールアドレスの形式が正しくありません。',
 *  'validation.required': '{0}を入力してください。',
 *  'validation.min': '{0}は{1}文字以上の値を入力してください。',
 * } as const;
 * ```
 * @example
 * メッセージを取得する例
 * ```
 * m('validation.email') // メールアドレスの形式が正しくありません。
 * m('validation.required', 'パスワード') // パスワードを入力してください。
 * m('validation.min', 'パスワード', '8') // パスワードは8文字以上の値を入力してください。
 * ```
 */
function message(key: MessageKey, ...options: string[]): string {
  if (!cache) {
    throw new Error('Messages are not loaded.');
  }
  if (cache[key] === undefined) {
    handleError(new RuntimeError(`Could not find the message. messageKey=[${key}]`, 'MessageNotFound'));
    return key;
  }
  return !options.length ? cache[key] : format(cache[key], options);
}

/**
 * メッセージをフォーマットします。
 * @param message メッセージ
 * @param options メッセージのオプション
 * @returns フォーマット後のメッセージ
 */
function format(message: string, options: string[]): string {
  return options.reduce((message, option, index) => {
    return message.replace(new RegExp(`\\{${index}\\}`, 'g'), option);
  }, message);
}

export type {MessagesLoader};
export {message as m, loadMessages};
