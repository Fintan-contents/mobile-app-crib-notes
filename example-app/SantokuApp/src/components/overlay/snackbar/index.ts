export * from './Snackbar';
export * from './WithSnackbar';
