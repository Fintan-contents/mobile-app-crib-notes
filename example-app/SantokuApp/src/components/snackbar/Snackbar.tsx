import React, {useCallback, useRef, useState} from 'react';
import {Animated, StyleProp, StyleSheet, Text, TextStyle, TouchableOpacity, View, ViewStyle} from 'react-native';

import {FullWindowOverlay} from './FullWindowOverlay';

import CompositeAnimation = Animated.CompositeAnimation;

export type SnackbarShowProps = {
  message: string;
  messageTextStyle?: StyleProp<TextStyle>;
  top?: number;
  bottom?: number;
  right?: number;
  left?: number;
  style?: StyleProp<ViewStyle>;
  animatedContainerStyle?: StyleProp<ViewStyle>;
  actionText?: string;
  actionHandler?: () => void;
  actionTextStyle?: StyleProp<TextStyle>;
  autoHideDuration?: number;
  fadeInDuration?: number;
  fadeOutDuration?: number;
  forceFadeOutDuration?: number;
};

export type SnackbarHideProps = {
  hide?: true;
  hideFadeOutDuration?: number;
};

export type SnackbarProps = SnackbarShowProps & SnackbarHideProps;

export const Snackbar: React.FC<SnackbarProps> = (props) => {
  const fadeAnim = useRef(new Animated.Value(0)).current;
  const fadeInAnimationRef = useRef<CompositeAnimation>();
  const fadeOutAnimationRef = useRef<CompositeAnimation>();
  const barrierFadeOutAnimationRef = useRef<CompositeAnimation>();
  const [visibleSnackbarProps, setVisibleSnackbarProps] = useState<SnackbarShowProps>();

  const animationStart = useCallback(
    (
      animationRef: React.MutableRefObject<CompositeAnimation | undefined>,
      config: Animated.TimingAnimationConfig,
      callback?: Animated.EndCallback,
    ) => {
      const animation = Animated.timing(fadeAnim, config);
      animationRef.current = animation;
      animation.start((result) => {
        animationRef.current = undefined;
        if (callback) {
          callback(result);
        }
      });
    },
    [fadeAnim],
  );

  const show = useCallback(() => {
    setVisibleSnackbarProps(props);

    const fadeInAnimationConfig = {
      toValue: 1,
      duration: props.fadeInDuration,
      useNativeDriver: true,
    };
    animationStart(fadeInAnimationRef, fadeInAnimationConfig, ({finished}) => {
      if (finished) {
        const fadeOutAnimationConfig = {
          toValue: 0,
          delay: props.autoHideDuration,
          duration: props.fadeOutDuration,
          useNativeDriver: true,
        };
        animationStart(fadeOutAnimationRef, fadeOutAnimationConfig, () => {
          if (!barrierFadeOutAnimationRef) {
            setVisibleSnackbarProps(undefined);
          }
        });
      } else {
        if (!barrierFadeOutAnimationRef) {
          setVisibleSnackbarProps(undefined);
        }
      }
    });
  }, [props, animationStart]);

  const forceFadeout = useCallback(
    (fadeOutDuration?: number, callback?: () => void) => {
      const barrierFadeOutAnimationConfig = {
        toValue: 0,
        duration: fadeOutDuration,
        useNativeDriver: true,
      };

      fadeInAnimationRef.current?.stop();
      fadeOutAnimationRef.current?.stop();

      animationStart(barrierFadeOutAnimationRef, barrierFadeOutAnimationConfig, () => {
        setVisibleSnackbarProps(undefined);
        callback?.();
      });
    },
    [animationStart],
  );

  React.useEffect(() => {
    if (props.hide) {
      forceFadeout(props.hideFadeOutDuration ?? props.forceFadeOutDuration);
      return;
    }
    if (!props.message) {
      return;
    }
    if (barrierFadeOutAnimationRef.current) {
      return;
    }
    if (fadeInAnimationRef.current || fadeOutAnimationRef.current) {
      forceFadeout(props.forceFadeOutDuration, show);
      return;
    }
    show();
  }, [props, show, forceFadeout]);

  const snackbarStyle = StyleSheet.flatten([styles.snackbar, props.style]);

  const animatedViewStyle = StyleSheet.flatten<ViewStyle>([styles.animatedContainer, props.animatedContainerStyle]);
  if (animatedViewStyle?.top === undefined && animatedViewStyle.bottom === undefined) {
    animatedViewStyle.bottom = 20;
  }

  return (
    <>
      {props.children}
      {visibleSnackbarProps && (
        <FullWindowOverlay>
          <Animated.View style={StyleSheet.flatten([{opacity: fadeAnim}, animatedViewStyle])}>
            <View style={snackbarStyle}>
              <View style={styles.messageContainer}>
                <Text style={styles.messageText}>{visibleSnackbarProps.message}</Text>
              </View>
              {visibleSnackbarProps.actionText && visibleSnackbarProps.actionHandler && (
                <View style={styles.actionContainer}>
                  <TouchableOpacity onPress={visibleSnackbarProps.actionHandler}>
                    <Text style={styles.actionText}>{visibleSnackbarProps.actionText}</Text>
                  </TouchableOpacity>
                </View>
              )}
            </View>
          </Animated.View>
        </FullWindowOverlay>
      )}
    </>
  );
};

Snackbar.defaultProps = {
  autoHideDuration: 4000,
  fadeInDuration: 1000,
  fadeOutDuration: 1000,
  forceFadeOutDuration: 300,
};

const styles = StyleSheet.create({
  animatedContainer: {
    flex: 1,
    position: 'absolute',
    minHeight: 48,
    paddingHorizontal: 10,
    width: '100%',
  },
  snackbar: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 5,
    backgroundColor: '#393939',
  },
  messageContainer: {
    flex: 1,
    paddingVertical: 20,
    paddingLeft: 20,
    paddingRight: 10,
  },
  actionContainer: {
    flex: 0,
    justifyContent: 'flex-end',
    alignItems: 'center',
    paddingVertical: 20,
    paddingLeft: 10,
    paddingRight: 20,
  },
  messageText: {
    color: 'white',
  },
  actionText: {
    color: '#85A3F1',
  },
});
