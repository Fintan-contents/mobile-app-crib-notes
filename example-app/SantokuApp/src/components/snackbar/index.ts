export * from './FullWindowOverlay';
export * from './Snackbar';
export * from './WithSnackbar';
