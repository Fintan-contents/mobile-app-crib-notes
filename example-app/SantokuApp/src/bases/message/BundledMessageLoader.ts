import {bundledMessages, BundledMessagesType} from './BundledMessages';
import {MessagesLoader} from './Message';

/**
 * アプリ内にバンドルされたメッセージをロードするクラスです。
 */
class BundledMessagesLoader implements MessagesLoader {
  load() {
    return new Promise<BundledMessagesType>(resolve => resolve(bundledMessages));
  }
}

export {BundledMessagesLoader};
