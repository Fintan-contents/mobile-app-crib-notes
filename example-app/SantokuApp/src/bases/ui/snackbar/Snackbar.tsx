import {m} from 'bases/message/Message';
import React, {useMemo, useState} from 'react';

import {
  SnackbarComponent,
  SnackbarHideProps as SnackbarComponentHideProps,
  SnackbarProps as SnackbarComponentProps,
  SnackbarShowProps as SnackbarComponentShowProps,
} from './SnackbarComponent';
import {FullWindowOverlay} from '../overlay/FullWindowOverlay';

type SnackbarShowProps = Omit<SnackbarComponentShowProps, 'message'>;
type SnackbarShowCloseButtonProps = Omit<SnackbarShowProps, 'actionText' | 'actionHandler' | 'actionTextStyle'>;
type SnackbarHideProps = Omit<SnackbarComponentHideProps, 'hide'>;

type SnackbarType = {
  /**
   * Show snackbar.
   *
   * @param message - Displayed message.
   * @param showProps - Snackbar props(SnackbarShowContextProps).
   */
  show: (message: string, showProps?: SnackbarShowProps) => void;

  /**
   * Show the snackbar with the close button.
   * The close button is placed to the right of the message.
   *
   * @param message - Displayed message.
   * @param showProps - Snackbar props(SnackbarShowCloseButtonContextProps).
   */
  showWithCloseButton: (message: string, showProps?: SnackbarShowCloseButtonProps) => void;

  /**
   * Hide snackbar.
   *
   * @param hideProps - Snackbar props(SnackbarHideContextProps).
   */
  hide: (hideProps?: SnackbarHideProps) => void;
  Component: typeof Component;
};

let Snackbar: SnackbarType = {
  show: () => {
    throw new Error('Snackbar.Component is required.');
  },
  showWithCloseButton: () => {
    throw new Error('Snackbar.Component is required.');
  },
  hide: () => {
    throw new Error('Snackbar.Component is required.');
  },
  Component,
};

function Component(props: {initialState?: SnackbarComponentShowProps}) {
  const [state, setState] = useState<SnackbarComponentProps>(
    props.initialState ?? {
      message: '',
    },
  );
  // const snackbarContext = useMemo<SnackbarContextType>(
  Snackbar = useMemo<SnackbarType>(
    () => ({
      ...Snackbar,
      show: (message: string, showProps?: SnackbarShowProps) => {
        setState({
          timestamp: Date.now(),
          ...showProps,
          message,
        });
      },
      showWithCloseButton: (message: string, showProps?: SnackbarShowCloseButtonProps) => {
        setState({
          timestamp: Date.now(),
          ...showProps,
          actionText: m('閉じる'),
          actionHandler: Snackbar.hide,
          message,
        });
      },
      hide: (hideProps?: SnackbarHideProps) => {
        setState(prevState => ({...prevState, ...hideProps, hide: true}));
      },
    }),
    [],
  );
  return (
    <>
      <FullWindowOverlay>
        <SnackbarComponent {...state} />
      </FullWindowOverlay>
    </>
  );
}

export {Snackbar};
