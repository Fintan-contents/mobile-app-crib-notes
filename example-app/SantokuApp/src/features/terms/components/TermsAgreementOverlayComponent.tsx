import {m} from 'bases/message/Message';
import {Button} from 'bases/ui/button/Button';
import {OverlayBackdrop} from 'bases/ui/overlay/OverlayBackdrop';
import {OverlayContainer} from 'bases/ui/overlay/OverlayContainer';
import {WebView} from 'bases/ui/webview/WebView';
import {useIsLoggedIn} from 'features/account/client-states/useIsLoggedIn';
import {useAccountCommands} from 'features/account/services/account/useAccountCommands';
import {TermsOfServiceAgreementStatus} from 'features/backend/apis/model';
import {TermsOfService} from 'features/backend/apis/model/termsOfService';
import React, {useCallback, useEffect, useMemo, useState} from 'react';
import {StyleSheet, View} from 'react-native';
import {Text} from 'react-native-elements';

export type TermsAgreementOverlayShowProps = {
  termsOfService: TermsOfService;
  /**
   * Overlayの背景をタップした時に閉じるかどうかの設定。
   * デフォルトはtrue（閉じる）です。
   */
  dismissible?: boolean;
  /**
   * iOSの場合、アニメーションが終わった後に呼び出されます。
   * Androidの場合、アニメーションが始まった時に呼び出されます。
   */
  enteringCallback?: (finished: boolean) => unknown;
  exitingCallback?: (finished: boolean) => unknown;
  exitingCallbackOnAgreed?: (termsOfServiceAgreementStatus: TermsOfServiceAgreementStatus) => unknown;
};

export type TermsAgreementOverlayProps = TermsAgreementOverlayShowProps & {
  visible: boolean;
  close: () => void;
  contentViewTestID?: string;
};

export const TermsAgreementOverlayComponent: React.FC<TermsAgreementOverlayProps> = ({
  visible,
  close,
  termsOfService,
  dismissible = true,
  enteringCallback,
  exitingCallback,
  exitingCallbackOnAgreed,
  contentViewTestID,
}) => {
  const [isWebViewError, setIsWebViewError] = useState(false);
  const [buttonDisable, setButtonDisable] = useState(true);
  const [isExited, setIsExited] = useState(false);
  const [agreedStatus, setAgreedStatus] = useState<TermsOfServiceAgreementStatus>();

  const [isLoggedIn] = useIsLoggedIn();
  const {agreeTerms, isAgreeingTerms} = useAccountCommands();

  const webViewSource = useMemo(() => {
    const termUrl = termsOfService?.url;
    return termUrl ? {uri: termUrl} : undefined;
  }, [termsOfService?.url]);

  const onWebViewError = useCallback(() => {
    setIsWebViewError(true);
  }, [setIsWebViewError]);

  const resetWebViewError = useCallback(() => {
    setIsWebViewError(false);
  }, [setIsWebViewError]);

  const composedExitingCallback = useCallback(
    (finished: boolean) => {
      try {
        exitingCallback?.(finished);
      } finally {
        setButtonDisable(true);
        setIsExited(true);
      }
    },
    [exitingCallback, setButtonDisable, setIsExited],
  );

  const onScrollEndOnce = useCallback(() => setButtonDisable(false), [setButtonDisable]);

  const onAgree = useCallback(async () => {
    try {
      const termsAgreementStatus = {
        hasAgreed: true,
        agreedVersion: termsOfService.version,
      };
      if (isLoggedIn) {
        await agreeTerms(termsAgreementStatus);
      }
      setAgreedStatus(termsAgreementStatus);
      close();
    } catch {
      // 個別のエラーハンドリングは不要
    }
  }, [agreeTerms, close, isLoggedIn, setAgreedStatus, termsOfService.version]);

  const exitOnAgreed = useCallback(() => {
    if (isExited && agreedStatus) {
      try {
        exitingCallbackOnAgreed?.(agreedStatus);
      } finally {
        setIsExited(false);
        setAgreedStatus(undefined);
      }
    }
  }, [agreedStatus, exitingCallbackOnAgreed, isExited, setAgreedStatus, setIsExited]);

  useEffect(() => exitOnAgreed(), [exitOnAgreed]);

  return (
    <OverlayBackdrop
      isVisible={visible}
      onPress={dismissible ? close : undefined}
      enteringCallback={enteringCallback}
      exitingCallback={composedExitingCallback}>
      <OverlayContainer isVisible={visible} style={styles.modalContainer}>
        <View style={styles.container} testID={contentViewTestID}>
          <View style={styles.header}>
            <Text style={styles.headerText}>{m('利用規約')}</Text>
          </View>
          {isWebViewError ? (
            <View style={styles.errorView}>
              <Text style={styles.errorText}>{m('app.terms.有効な利用規約の取得エラー')}</Text>
              <Button title={m('リロード')} size="full" onPress={resetWebViewError} />
            </View>
          ) : (
            <WebView
              source={webViewSource}
              onScrollEndOnce={onScrollEndOnce}
              onError={onWebViewError}
              onHttpError={onWebViewError}
            />
          )}
          <View style={styles.footer}>
            <Button title={m('同意')} onPress={onAgree} disabled={buttonDisable || isAgreeingTerms} />
          </View>
        </View>
      </OverlayContainer>
    </OverlayBackdrop>
  );
};

const styles = StyleSheet.create({
  modalContainer: {
    paddingHorizontal: 15,
    paddingVertical: 50,
  },
  container: {
    height: '100%',
    width: '100%',
    justifyContent: 'center',
    backgroundColor: 'white',
    paddingHorizontal: 10,
    paddingVertical: 20,
    borderRadius: 10,
  },
  header: {
    alignSelf: 'center',
    paddingVertical: 10,
  },
  headerText: {
    fontSize: 24,
    fontWeight: 'bold',
  },
  errorView: {
    flex: 1,
    padding: 20,
    alignItems: 'center',
    justifyContent: 'center',
  },
  errorText: {
    fontSize: 14,
    marginBottom: 20,
  },
  footer: {
    flex: 0,
    flexDirection: 'row-reverse',
    paddingVertical: 14,
    paddingHorizontal: 20,
  },
});
