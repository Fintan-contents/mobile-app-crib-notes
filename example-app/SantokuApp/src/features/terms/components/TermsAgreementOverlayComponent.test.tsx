import '@testing-library/jest-native/extend-expect';
import {render, screen} from '@testing-library/react-native';
import {AppConfig} from 'bases/core/configs/AppConfig';
import {BundledMessagesLoader} from 'bases/message/BundledMessageLoader';
import {loadMessages} from 'bases/message/Message';
import {Snackbar} from 'bases/ui/snackbar/Snackbar';
import React from 'react';
import {SafeAreaProvider} from 'react-native-safe-area-context';
import {QueryClient, QueryClientProvider} from 'react-query';

import {TermsAgreementOverlayComponent} from './TermsAgreementOverlayComponent';

const Wrapper: React.FC<React.PropsWithChildren> = ({children}) => {
  const queryClient = new QueryClient();
  return (
    <SafeAreaProvider>
      <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
      <Snackbar.Component />
    </SafeAreaProvider>
  );
};

beforeAll(async () => {
  await loadMessages(new BundledMessagesLoader());
});

describe('TermsAgreementOverlayComponent', () => {
  it('マウントされたときに正常にレンダリングされること', () => {
    // importでLoginScreenを読み込むと、メッセージのロードが完了する前にメッセージを読み込んでしまうため、requireで取得する
    // requireした場合の型はanyとなってしまいESLintエラーが発生しますが無視します。
    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
    render(
      <TermsAgreementOverlayComponent
        visible
        close={() => {}}
        termsOfService={{version: '1.0.0', url: AppConfig.termsUrl}}
        contentViewTestID="TermsAgreementOverlay"
      />,
      {
        wrapper: Wrapper,
      },
    );
    expect(screen.queryByTestId('TermsAgreementOverlay')).not.toBeNull();
    expect(screen).toMatchSnapshot();
  });
});
