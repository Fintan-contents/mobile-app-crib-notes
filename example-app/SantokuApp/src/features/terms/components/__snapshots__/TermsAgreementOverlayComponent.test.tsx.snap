// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TermsAgreementOverlayComponent マウントされたときに正常にレンダリングされること 1`] = `
Array [
  <View
    style={
      Object {
        "bottom": 0,
        "height": 640,
        "position": "absolute",
        "width": 320,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        animatedStyle={
          Object {
            "value": Object {},
          }
        }
        collapsable={false}
        entering={
          FadeIn {
            "build": [Function],
            "callbackV": [Function],
            "durationV": 300,
            "randomizeDelay": false,
          }
        }
        exiting={
          FadeOut {
            "build": [Function],
            "callbackV": [Function],
            "durationV": 300,
            "randomizeDelay": false,
          }
        }
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.4)",
            "flex": 1,
          }
        }
      />
    </View>
    <View
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-end",
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        }
      }
    >
      <View
        animatedStyle={
          Object {
            "value": Object {},
          }
        }
        collapsable={false}
        entering={
          FadeIn {
            "build": [Function],
            "callbackV": [Function],
            "durationV": 300,
            "randomizeDelay": false,
          }
        }
        exiting={
          FadeOut {
            "build": [Function],
            "callbackV": [Function],
            "durationV": 300,
            "randomizeDelay": false,
          }
        }
        pointerEvents="box-none"
        style={
          Object {
            "paddingHorizontal": 15,
            "paddingVertical": 50,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 10,
              "height": "100%",
              "justifyContent": "center",
              "paddingHorizontal": 10,
              "paddingVertical": 20,
              "width": "100%",
            }
          }
          testID="TermsAgreementOverlay"
        >
          <View
            style={
              Object {
                "alignSelf": "center",
                "paddingVertical": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#242424",
                  "fontSize": 24,
                  "fontWeight": "bold",
                }
              }
            >
              利用規約
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "overflow": "hidden",
                },
                undefined,
              ]
            }
          >
            <RNCWebView
              androidLayerType="software"
              cacheEnabled={true}
              decelerationRate={0.998}
              injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
              injectedJavaScriptForMainFrameOnly={true}
              javaScriptEnabled={true}
              messagingEnabled={true}
              onContentProcessDidTerminate={[Function]}
              onHttpError={[Function]}
              onLoadingError={[Function]}
              onLoadingFinish={[Function]}
              onLoadingProgress={[Function]}
              onLoadingStart={[Function]}
              onMessage={[Function]}
              onScroll={[Function]}
              onShouldStartLoadWithRequest={[Function]}
              source={
                Object {
                  "uri": "https://www.tis.co.jp/termsofuse/",
                }
              }
              style={
                Array [
                  Object {
                    "flex": 1,
                    "overflow": "hidden",
                  },
                  Object {
                    "backgroundColor": "#ffffff",
                  },
                  Object {
                    "flex": 1,
                  },
                ]
              }
              textInteractionEnabled={true}
              useSharedProcessPool={true}
            />
            <ActivityIndicator
              color="#0000ff"
              size="large"
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "height": "100%",
                  "justifyContent": "center",
                  "position": "absolute",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "flex": 0,
                "flexDirection": "row-reverse",
                "paddingHorizontal": 20,
                "paddingVertical": 14,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderRadius": 3,
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 3,
                  },
                  Object {
                    "borderRadius": 8,
                    "width": 90,
                  },
                  false,
                ]
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "busy": false,
                    "disabled": true,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "hsl(208, 8%, 90%)",
                      "borderColor": "#2089dc",
                      "borderRadius": 8,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "height": 50,
                      "justifyContent": "center",
                      "padding": 8,
                      "width": 90,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "hsl(208, 8%, 63%)",
                        "fontSize": 18,
                        "paddingVertical": 1,
                        "textAlign": "center",
                      }
                    }
                  >
                    同意
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>,
  <View
    testID="FullWindowOverlay"
  />,
]
`;
